### PR TITLE
Fix serialize for responding custom JSON content-types

### LINKED ISSUE #753: Cannot set correct Content-Type header for JSON API
The following works fine:
```
request.set('Content-Type', 'application/json')
```
whereas using the JSON API header as follows:
```
request.set('Content-Type', 'application/vnd.api+json')
```
results in the following error:
```
TypeError: first argument must be a string or Buffer
      at ClientRequest.OutgoingMessage.end (_http_outgoing.js:517:11)
      at Test.Request.end (node_modules/supertest/node_modules/superagent/lib/node/index.js:1020:9)
      at Test.end (node_modules/supertest/lib/test.js:121:7)
      at Context.<anonymous> (test/create.test.js:39:10)
```

Bypassing superagent and using CURL works fine:
```
curl -X POST -H "Content-Type: application/vnd.api+json" --data  '{"data": { "type": "posts", "attributes": { "title": "test", "content": "test content" } }}' http://127.0.0.1:3000/api/posts
```

### PR DESCRIPTION
**Pull request by Eiryyy**
Fixed #753

### CODE REVIEW

**Comment by Eiryyy** [comment]
Does this duplicate with #695 ?

**Comment by listepo** [comment]
+1

**Comment by listepo** [comment]
@defunctzombie review this PR please

**Comment by flipxfx** [comment]
+1

**Comment by focusaurus** [comment]
LGTM. @pornel can you review please?

**Comment by heathprovost** [comment]
+1

File: lib/client.js [hunk 1]
```diff
@@ -220,6 +220,22 @@ function parseHeader(str) {
...
+ * @return {Boolean}
+ * @api private
+ */
+
+function isJSON(mime) {
+  if (!mime) return false;
+  var parts = mime.split('/');
+  var type = parts[0];
+  var subtype = parts[1];
+  return subtype && subtype.match(/json/i);
```
Comment (pornel) [inline lib/client.js]: Since you're using regex anyway, you could simplify it as `/[\\+]json\b/.test(mime)`

Comment (Eiryyy) [inline lib/client.js]: The collect is not `/[\\+]json\b/.test(mime)` but `/[\/+]json\b/.test(mime)`, I suppose. And, why `\b` (not`$`)?

Comment (pornel) [inline lib/client.js]: Right, I've got slashes backwards.

`\b` to support `foo/bar+json;charset=x`

**Comment by Eiryyy** [comment]
@pornel Thank you. Rereview please

**Comment by pornel** [comment]
Thank you

**Pull request merged**